src/main/java/org/apache/mesos/hadoop/Utils.java

```diff
@@ -34,4 +40,96 @@ public static ByteString confToBytes(Configuration conf) throws IOException {
     byte[] bytes = baos.toByteArray();
     return ByteString.copyFrom(bytes);
   }
+
+  public static CommandInfo.ContainerInfo buildContainerInfo(Configuration conf) {
+    String containerImage = conf.get("mapred.mesos.container.image");
+    String[] containerOptions = conf.getStrings("mapred.mesos.container.options");
+
+    CommandInfo.ContainerInfo.Builder containerInfo =
+        CommandInfo.ContainerInfo.newBuilder();
+
+    if (containerImage != null) {
+      containerInfo.setImage(containerImage);
+    }
+
+    if (containerOptions != null) {
+      for (int i = 0; i < containerOptions.length; i++) {
+        containerInfo.addOptions(containerOptions[i]);
+      }
+    }
+
+    return containerInfo.build();
+  }
+
+  public static ContainerInfo buildDockerContainerInfo(Configuration conf) {
+    ContainerInfo.Builder containerInfoBuilder = ContainerInfo.newBuilder();
+    ContainerInfo.DockerInfo.Builder dockerInfoBuilder = ContainerInfo.DockerInfo.newBuilder();
+
+    dockerInfoBuilder.setImage(conf.get("mapred.mesos.docker.image"));
+
+    switch (conf.getInt("mapred.mesos.docker.network", 1)) {
+      case 1:
+        dockerInfoBuilder.setNetwork(ContainerInfo.DockerInfo.Network.HOST);
+      case 2:
+        dockerInfoBuilder.setNetwork(ContainerInfo.DockerInfo.Network.BRIDGE);
+      case 3:
+        dockerInfoBuilder.setNetwork(ContainerInfo.DockerInfo.Network.NONE);
+    }
+
+    dockerInfoBuilder.setPrivileged(conf.getBoolean("mapred.mesos.docker.privileged", false));
+    dockerInfoBuilder.setForcePullImage(conf.getBoolean("mapred.mesos.docker.force_pull_image", false));
+
+    // Parse out any additional docker CLI params
+    String[] params = conf.getStrings("mapred.mesos.docker.parameters");
+    if (params.length > 0) {
+      assert (params.length % 2) == 0; // Make sure we have an even number of parameters
```

Comment:
Iirc `assert()` will be ignored if debug is disabled, I'd rather have the check thrown an `IllegalArgumentException`, personally.